### PR TITLE
Print build-time warnings for LGPL gotchas

### PR DESCRIPTION
**Pull request by grdanny**
Trying out my first Mackerel from:

https://github.com/OpenImageIO/oiio/issues/3916

### CODE REVIEW

File: src/ffmpeg.imageio/CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,34 @@
...
+
+    if (${_static_libraries_found} GREATER 0)
+     message("${ColorRed}")
+     message(STATUS "Looks like you are linking against some static ffmpeg libraries: ${FFMPEG_LIBRARIES}")
+     message(STATUS "ffmpeg is a LGPL-licensed library")
+     message(STATUS "When linking statically against a LGPL-licensed library your entire project needs to be under LGPL")
+     message("${ColorReset}")
+    endif()
+
+   endif()
```
Comment (lgritz) [inline src/ffmpeg.imageio/CMakeLists.txt:30]: I like the approach. Danny, can you please clean up the indentation -- they don't always align, and some blocks intend 1 space, some 2, some 4? Let's use a consistent 4 like the rest of the cmake files.

File: src/heif.imageio/CMakeLists.txt [hunk 2]
```diff
@@ -3,6 +3,33 @@
 # https://github.com/OpenImageIO/oiio
 
 if (Libheif_FOUND)
+   if (LINKSTATIC)
+    if (WIN32)
+        set (_static_suffixes .lib .a)
+    else ()
+        set (_static_suffixes .a)
+    endif ()
```
Comment (lgritz) [inline src/heif.imageio/CMakeLists.txt:11]: Can this be simplified? If you just do .lib and .a, and not need the `if`, so what? It won't find .lib in the libraries on non-Windows, so it doesn't hurt to search for it.

File: src/heif.imageio/CMakeLists.txt [hunk 3]
```diff
@@ -3,6 +3,33 @@
...
+       if (${_index} GREATER -1)
+         MATH(EXPR _static_libraries_found "${static_libraries_found}+1")
+       endif()
+    endforeach()
+
+    if (${_static_libraries_found} GREATER 0)
+     message("${ColorRed}")
+     message(STATUS "Looks like you are linking against some static libheif libraries")
+     message(STATUS "libheif is a LGPL-licensed library")
+     message(STATUS "When linking statically against a LGPL-licensed library your entire project needs to be under LGPL")
```
Comment (lgritz) [inline src/heif.imageio/CMakeLists.txt:26]: Let's omit the prescriptive legal advice and go with a simpler informational message that leaves the problem on their doorstep:

```
You are linking OpenImageIO against a static version of libheif, which is LGPL
licensed. If you intend to redistribute this build of OpenImageIO, we recommend
that you review the libheif license terms, or you may wish to switch to using a
dynamically-linked libheif.
```

I think this gives them the information they need to point them in the right direction, and avoids letting them stumble unaware into possible trouble, but without trying to tell them exactly what they can and can't do (which, as non-lawyers, we should not be doing and could be wrong about).

A slightly different message more accurately covers the ffmpeg situation:

```
You are linking OpenImageIO against a static version of FFmpeg, which may have
LGPL and possibly GPL licensed components (depending on exactly how your copy
of FFmpeg was built). If you intend to redistribute this build of OpenImageIO,
we recommend that you review the FFmpeg build flags and licensing terms.
```

**Review by lgritz — CHANGES_REQUESTED** [review]
I think you're on the right track to fully solve this issue. I recommended a few minor changes.

File: src/ffmpeg.imageio/CMakeLists.txt [hunk 4]
```diff
@@ -3,6 +3,34 @@
...
+    foreach(_ffmpeg_library IN LISTS FFMPEG_LIBRARIES)
+       get_filename_component(_ext ${_ffmpeg_library} LAST_EXT)
+       list (FIND _static_suffixes ${_ext} _index)
+       if (${_index} GREATER -1)
+         MATH(EXPR _static_libraries_found "${static_libraries_found}+1")
+       endif()
+    endforeach()
+
+    if (${_static_libraries_found} GREATER 0)
+     message("${ColorRed}")
```
Comment (lgritz) [inline src/ffmpeg.imageio/CMakeLists.txt:23]: Let's use ColorYellow here. It's not a build break, just a warning.

**Comment by grdanny** [comment]
This should be ready for review.
The only additional thought I had was maybe to turn this into a macro and share across the two cases (and maybe then it's useful in some other similar future case?)
But not sure if that is more helpful or confusing, right now it should be pretty clear why we added this.

**Review by lgritz — APPROVED** [review]

**Comment by lgritz** [comment]
I think it's fine as it is, Danny. These are the only two instances of this that we know about. If another pops up, we can think about whether to refactor into some common infrastructure.

**Comment by lgritz** [comment]
Fixes #3916

**Pull request merged**